web/source/osk/visualKeyboard.ts

```diff
@@ -172,53 +183,46 @@ namespace com.keyman.osk {
     }
 
     public postInsert(): void { }
-
+ 
+    /**
+     * The configured width for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic width scaling. 
+     */
     get width(): number {
-      if(this._width) {
-        return this._width;
-      } else {
-        this.loadSizeFromCookie();
-        return this.width;
-      }
+      return this._width;
     }
 
+    /**
+     * The configured height for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic height scaling. 
+     */
     get height(): number {
-      if(this._height) {
-        return this._height;
-      } else {
-        this.loadSizeFromCookie();
-        return this.height;
-      }
+      return this._height;
     }
 
-    protected loadSizeFromCookie() {
-      let keyman = com.keyman.singleton;
-      let util = keyman.util;
-
-      // If no prior cookie exists, it merely returns an empty object / cookie.
-      var c = util.loadCookie('KeymanWeb_OnScreenKeyboard');
-      var newWidth: number, newHeight: number;
-
-      // Restore OSK size - font size now fixed in relation to OSK height, unless overridden (in em) by keyboard
-      newWidth=util.toNumber(c['width'], 0.333 * screen.width); // Default - 1/3rd of screen's width.
+    get fontSize(): ParsedLengthStyle {
+      return this._fontSize;
+    }
 
-      if(newWidth < 0.2*screen.width) {
-        newWidth = 0.2*screen.width;
-      } else if(newWidth > 0.9*screen.width) {
-        newWidth=0.9*screen.width;
-      }
-  
-      // Default height decision made here: 
-      // https://github.com/keymanapp/keyman/pull/4279#discussion_r560453929
-      newHeight=util.toNumber(c['height'], 0.333 * newWidth);
+    set fontSize(value: ParsedLengthStyle) {
+      this._fontSize = value;
+      this.kbdDiv.style.fontSize = value.styleString;
+    }
```

Comment:
1.  We don't automatically `refreshLayout()` here because the `OSKManager` controls when that happens.  Also, because this can be but one call of many that affects the layout; `OSKManager` will delay until all `VisualKeyboard` properties are updated, then do the layout refresh.
2. It was always weird that desktop keyboard font sizes were set here while touch keyboard sizes were set in `OSKManager`.  (See code removed from its `_Load`.)
3. By making it a mutable property on `VisualKeyboard`, it can also be made a mutable property on `OSKManager` in the future, facilitating a decent feature.